### ISSUE TITLE
[Feature Request] Scrap only first N comments.

### ISSUE THREAD

**Issue by dajotim937** [instaloader/instaloader#1391]
I saw feature request to scrap top N comments. And I understand that for this you still need fetch all comments. But can we fetch only first N comments?
My situation:
I saved posts via bookmark feature and now downloading those posts with comments. Those posts are usually some movie trailer or pic from movie. And not everyone write movie title in description but usually users asks in comments and someone answer to them. That why I donwload comments too. BUT. Sometimes post has like 1k+ comments and movie title in first 250 comments max, so I don't need other 750 comments but I'm still waiting until instaloader fetch them all. Can I somehow limit number of fetching comments?

**Comment by dajotim937** [comment]
Hi. Related issue:
I have this post `/p/CYLcqzOok8a` in saved. But for some reason, instaloader I have this output when you are downloading it:
```
saved\some.pictures.of.her_2022-01-01_CYLcqzOok8a.jpg exists [Photo shoot with Camille, Par…] unchanged Resuming from ：saved\some.pictures.of.her_2022-01-01_CYLcqzOok8a_iterator_uUToSNpz.json.xz.
13000…13250…13500…13750…14000…14250…14500…14750…15000…15250…15500…15750…16000…16250…16500…16750…17000…17250…17500…17750…
```
Using iterator because it's second day that I tried to download saved posts.

After so many comments I downloaded only image with metadata.  In metada file I found that field "comment_count" has value about 150.
After that I used this command to recreate issue:
`instaloader.exe --sessionfile="<path>" --login=<login> --comments -- -CYLcqzOok8a` and by output instaloader fetched 500 comments before I abort it.
<details>
<summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/96221151/151630879-5d3c96c9-3919-466d-bc9e-96b407d9e26c.png)
</details>

After abort `2022-01-01_07-21-05_UTC_comments.json` file was created with all comments and also iterator file was created. In iterator json file there is field
```
 "remaining_data": {
            "count": 152,
```
But all 152 comments already downloaded. Should I create new issue for this bug?

**Comment by dajotim937** [comment]
@Thammus Any comments? Why this was closed?

**Comment by Thammus** [comment]
This issue was closed because you did not fill out the issue template which contains merely the basic questions that we always have to ask when deciding how to proceed with a feature suggestion.

Limiting the number of downloaded comments is not possible through the CLI and is not considered a relevant feature. This can easily be achieved using the iterator returned by [get_comments()](https://instaloader.github.io/module/structures.html?highlight=get_comments#instaloader.Post.get_comments).

Additionally, your log that shows over 1700 comments while the post only seems to have about 150 indicates that you are running into the "endless loop when receiving comments" (#1347) bug which is fixed since Instaloader 4.8.2.

**Comment by dajotim937** [comment]
> Additionally, your log that shows over 1700 comments while the post only seems to have about 150 indicates that you are running into the "endless loop when receiving comments" (#1347) bug which is fixed since Instaloader 4.8.2.

Yeah. Fixed. Since 4.8.2. Right. 
First `--version` output was from version which I used before when reported bug. Then I updated instaloader.
![image](https://user-images.githubusercontent.com/96221151/153638050-e9cba3b2-9b65-4425-81a4-31681e086f89.png)
From iterator json:
![image](https://user-images.githubusercontent.com/96221151/153638271-df28512d-3052-4630-b875-74caa2088742.png)
From _comments json:
![image](https://user-images.githubusercontent.com/96221151/153638435-1306dd9d-3e94-4a03-a33c-dfe0d2f56768.png)

I guess 162-163 comments is less that 500 comments downloaded by instaloader in it cli output, what do you think?
